specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/examples/Patches_Apply.json

```diff
@@ -0,0 +1,88 @@
+{
+  "operationId": "Patches_Apply",
+  "title": "Patches_Apply_0",
+  "parameters": {
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "rg",
+    "builderName": "testBuilder",
+    "patchName": "testPatch-25fe4b",
+    "api-version": "2023-08-01-preview"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/builders/testBuilder/patches/testPatch-25fe4b",
+        "name": "testPatch-25fe4b",
+        "type": "Microsoft.App/builders/patches",
+        "properties": {
+          "targetEnvironmentId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/managedEnvironments/test-env",
+          "targetContainerAppId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/containerApps/test-app",
+          "targetRevisionId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/apps/test-app/revisions/test-app--jm3vvry",
+          "provisioningState": "Succeeded",
+          "patchApplyStatus": "Succeeded",
+          "createdTime": "2022-10-10T12:06:20.3421+00:00",
+          "updatedTime": "2022-10-10T12:06:20.3421+00:00",
+          "patchDetails": [
+            {
+              "targetContainerName": "test-container",
+              "targetImage": "testregistry.azurecr.io/test-image:latest",
+              "lastDetectionTime": "2022-10-10T12:06:19.5241+00:00",
+              "detectionStatus": "Succeeded",
+              "newImageName": "testregistry.azurecr.io/test-image:latest-patched-202210101206",
+              "newLayer": {
```

Comment:
>newLayer [](http://example.com/codeflow?start=15&length=8)

why is it called a layer anyway?

Reply:
Because the patching operation is trying to swap vulnerable layer of the image.